blockchain/subscription.go

```diff
@@ -19,28 +19,51 @@ type Subscription interface {
 // FOR TESTING PURPOSES //
 
 type MockSubscription struct {
```

Comment:
Isn't this `MockSubscription` now the same implementation as the `subscription` struct that is actually used. Could we just expose that `subscription` struct and use it in tests instead of having this mock?

Reply:
Mocked implementation has Push fn added, but also I'm not sure if we want to expose the real implementation(was thinking about that)

Reply:
Then I would say, make `MockSubscription` an extension of `subscription`, and remove all functions except `Push`.

```suggestion
type MockSubscription struct {
	*subscription
}

func NewMockSubscription() *MockSubscription {
	return &MockSubscription{
		subscription: &subscription{
			updateCh: make(chan *Event),
			closeCh:  make(chan void),
		},
	}
}

func (m *MockSubscription) Push(e *Event) {
	m.updateCh <- e
}
```

Reply:
fixed f973b3b